src/SFA.DAS.QnA.Application/Commands/ResetPagesToIncomplete/ResetPagesToIncompleteHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.QnA.Api.Types;
+using SFA.DAS.QnA.Api.Types.Page;
+using SFA.DAS.QnA.Application.Commands.SetPageAnswers;
+using SFA.DAS.QnA.Application.Services;
+using SFA.DAS.QnA.Data;
+
+namespace SFA.DAS.QnA.Application.Commands.ResetPagesToIncomplete
+{
+    public class ResetPagesToIncompleteHandler : SetAnswersBase, IRequestHandler<ResetPagesToIncompleteRequest, HandlerResponse<bool>>
+    {
+        private readonly ILogger<ResetPagesToIncompleteHandler> _logger;
+        public ResetPagesToIncompleteHandler(QnaDataContext dataContext, INotRequiredProcessor notRequiredProcessor, ITagProcessingService tagProcessingService, ILogger<ResetPagesToIncompleteHandler> logger) : base(dataContext, notRequiredProcessor, tagProcessingService, null)
+        {
+            _logger = logger;
+        }
+
+        public async Task<HandlerResponse<bool>> Handle(ResetPagesToIncompleteRequest request, CancellationToken cancellationToken)
+        {
+            var section = await _dataContext.ApplicationSections.SingleOrDefaultAsync(sec => sec.ApplicationId == request.ApplicationId && sec.SequenceNo == request.SequenceNo && sec.SectionNo == request.SectionNo, cancellationToken);
+            if (section == null) return new HandlerResponse<bool>(true);
```

Comment:
Should we throw an exception if section is not found?

Reply:
no it's just a check to reset a section, if a section is null, nothing to reset, so confirmation sent back